src/y-websocket.js

```diff
@@ -123,23 +123,49 @@ const readMessage = (provider, buf, emitSynced) => {
   return encoder
 }
 
+const base64_arraybuffer = async (data) => {
+  // Use a FileReader to generate a base64 data URI
+  const base64url = await new Promise((r) => {
+      const reader = new FileReader()
+      reader.onload = () => r(reader.result)
+      reader.readAsDataURL(new Blob([data]))
+  })
+
+  /*
+  The result looks like
+  "data:application/octet-stream;base64,<your base64 data>",
+  so we split off the beginning:
+  */
+  return base64url.substring(base64url.indexOf(',')+1)
+}
+
+const toText = (data) => {
+  return new TextDecoder("latin1").decode(data)
+}
+
 /**
  * @param {WebsocketProvider} provider
  */
 const setupWS = (provider) => {
   if (provider.shouldConnect && provider.ws === null) {
-    const websocket = new provider._WS(provider.url)
-    websocket.binaryType = 'arraybuffer'
+    const websocket = new provider._WS(provider.url, provider.protocols)
     provider.ws = websocket
     provider.wsconnecting = true
     provider.wsconnected = false
     provider.synced = false
 
     websocket.onmessage = (event) => {
+      var data = new Uint8Array(event.data);
+      if (!(event.data instanceof ArrayBuffer)) {
+        data = new Uint8Array(event.data.length);
+        data.forEach((_, i) => {
+          data[i] = event.data.charCodeAt(i);
+        });
+      }
       provider.wsLastMessageReceived = time.getUnixTime()
-      const encoder = readMessage(provider, new Uint8Array(event.data), true)
+      const encoder = readMessage(provider, data, true)
       if (encoding.length(encoder) > 1) {
-        websocket.send(encoding.toUint8Array(encoder))
+        websocket.send(toText(encoding.toUint8Array(encoder)))
```

Comment:
Why does this line send data as `latin1`-encoded whereas the other usages of `websocket.send()` send data as `base64`?

Reply:
sorry, this is from an early attempt, we shouldn't do that